aten/src/ATen/native/cpu/SparseFactories.cpp

```diff
@@ -66,9 +66,30 @@ void _spdiags_kernel_cpu(
       });
 }
 
+void _spdiags_backward_kernel_cpu(TensorIterator& iter, Tensor& grad_in) {
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND4(
+      at::ScalarType::BFloat16,
+      at::ScalarType::Half,
+      at::ScalarType::Bool,
+      at::ScalarType::ComplexHalf,
```

Comment:
No reason to support non-float/complex types:
```suggestion
  AT_DISPATCH_CASE_FLOATING_AND_COMPLEX_TYPES_AND3(
      at::ScalarType::BFloat16,
      at::ScalarType::Half,
      at::ScalarType::ComplexHalf,
```